spotbugs/src/main/java/edu/umd/cs/findbugs/filter/SourceMatcher.java

```diff
@@ -54,6 +55,16 @@ public boolean match(BugInstance bugInstance) {
             return false;
         }
         boolean result = fileName.match(bugFileName);
+        
+        // if no result try to match with the full path as well
+        if (!result && bugInstance.getPrimarySourceLineAnnotation().isSourceFileKnown()) {
+          bugFileName = bugInstance.getPrimarySourceLineAnnotation().getRealSourcePath();
+          if(bugFileName == null || bugFileName.isEmpty()){
+            return false;
```

Comment:
When we achieve this line, the debug log will be skipped. better to use:

```java
if(bugFileName != null && !bugFileName.isEmpty()){
  result = fileName.match(bugFileName);
}

Reply:
I actually refactored the whole method. The issue you pointed out, was applicable in two places. I think it should be betetr now.